src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -445,20 +451,27 @@ export class BundleDataClient {
           return false;
         }
 
+        const originSpokePoolClient = this.spokePoolClientManager.getSpokePoolClientByChainId(fill.originChainId);
         // If origin spoke pool client isn't defined, we can't validate it.
-        if (this.spokePoolClients[fill.originChainId] === undefined) {
+        if (!isDefined(originSpokePoolClient)) {
           return false;
         }
-        const matchingDeposit = this.spokePoolClients[fill.originChainId].getDeposit(fill.depositId);
+        const matchingDeposit = originSpokePoolClient.getDeposit(fill.depositId);
         const hasMatchingDeposit =
           matchingDeposit !== undefined && getRelayEventKey(fill) === getRelayEventKey(matchingDeposit);
         return hasMatchingDeposit;
       });
       await forEachAsync(fillsToCount, async (_fill) => {
-        const matchingDeposit = this.spokePoolClients[_fill.originChainId].getDeposit(_fill.depositId);
+        const originSpokePoolClient = this.spokePoolClientManager.getSpokePoolClientByChainId(_fill.originChainId);
+        const matchingDeposit = originSpokePoolClient?.getDeposit(_fill.depositId);
         assert(isDefined(matchingDeposit), "Deposit not found for fill.");
```

Comment:
This one gets a bit tricky to debug because the assert() will fire if `originSpokePoolClient` does not exist, OR if it does not contain the deposit that we were looking for. Additionally, the assert doesn't identify which origin chain or fill we were looking for.

(This was existing implementation, but wdyt about having two assert statements, and identifying the relevant chains & deposit IDs?)

```suggestion
        const originChain = getNetworkName(_fill.originChainId);
        const originSpokePoolClient = this.spokePoolClientManager.getSpokePoolClientByChainId(_fill.originChainId);
        assert(isDefined(originSpokePoolClient), `No SpokePoolClient for chain ${originChain}`);
        const matchingDeposit = originSpokePoolClient.getDeposit(_fill.depositId);
        assert(isDefined(matchingDeposit), `No ${originChain} deposit found for ${getNetworkName(_fill.destinationChainId)} fill ${_fill.depositId}`);
```